File: Ruby/line.rb
class Line

	def initialize
		@cells = []
	end

	def << (cell)
		@cells << cell
	end

	def removeHint(hint)
		@cells.each {|cell| cell.removeHint(hint)}
	end

end

__END__

class Line(object):

	def __init__(self):
		self.cells = []

	def removeHint(self, hint):
		for cell in self.cells:
			cell.removeHint(hint)

File: Java/src/main/java/com/clovisleoncio/kenkensolver/Line.java
package com.clovisleoncio.kenkensolver;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class Line {
	
	private List<Cell> cells = new ArrayList<>();

	public void add(Cell cell) {
		cells.add(cell);
	}

	public void removeHint(Integer value, Cell subject) {
		for (Cell cell : cells) {
			if (cell != subject) {
				cell.removeHint(value);
			}
		}
	}

}
